lib/extras/packed_image_convert.cc

```diff
@@ -59,16 +59,22 @@ Status ConvertPackedPixelFileToCodecInOut(const PackedPixelFile& ppf,
   io->metadata.m.animation.num_loops = ppf.info.animation.num_loops;
 
   // Convert the color encoding.
+  bool want_fallback_icc;
   if (!ppf.icc.empty()) {
     PaddedBytes icc;
     icc.append(ppf.icc);
-    if (!io->metadata.m.color_encoding.SetICC(std::move(icc))) {
-      fprintf(stderr, "Warning: error setting ICC profile, assuming SRGB");
-      io->metadata.m.color_encoding = ColorEncoding::SRGB(is_gray);
-    }
+    want_fallback_icc = !io->metadata.m.color_encoding.SetICC(std::move(icc));
   } else {
     JXL_RETURN_IF_ERROR(ConvertExternalToInternalColorEncoding(
         ppf.color_encoding, &io->metadata.m.color_encoding));
+    // Successful ConvertExternalToInternalColorEncoding does not mean there
+    // were no errors (and ICC was actually set).
+    want_fallback_icc = io->metadata.m.color_encoding.ICC().empty();
+  }
+
+  if (want_fallback_icc) {
+    fprintf(stderr, "Warning: error setting ICC profile, assuming SRGB");
+    io->metadata.m.color_encoding = ColorEncoding::SRGB(is_gray);
```

Comment:
Couldn't this condition be false because the color encoding *was* set, but not as an ICC? In that case this condition would do something wrong...
I believe we have a function to set the ICC from the internal colorencoding, perhaps we should use that?

Reply:
At the end of `ConvertExternalToInternalColorEncoding` makes an attempt to create ICC (`ColorEncoding::CreateICC` -> `ColorEncoding::MaybeCreateProfile`). But it fails (`AdoptWhitePoint` recognises invalid input).

So, if there is no ICC, it means it can not be created. And we need it for `CheckMetadata`.

Reply:
Fine for me. One sec, will send an update.